09_CalculatorStrategyCoR/src/main/java/bootcamp/data/Result.java

```diff
@@ -4,10 +4,25 @@
 import java.util.Optional;
 
 public class Result {
-    private final Status status = null; //FIXME
-    private final String statusMessage = null; //FIXME
-    private final Optional<BigDecimal> value = null; //FIXME
+    private final Status status;
+    private final String statusMessage;
+    private Optional<BigDecimal> value = Optional.empty();
 
-    //TODO Constructor. If there was an error in processing, set the relevant status, message and value to empty Optional.
-    //TODO Getters
-}
+    public Result(Status status, String statusMessage, Optional<BigDecimal> value) {
```

Comment:
Java authors recommend to avoid using Optional as parameter since by doing that you're increasing the work you do in the method verifying whether is present or not. Another good reason is that Optionals are conceived more as return types. As an alternative, you can use methods overload

Reply:
Good to know, I was following the Kata guidelines, but INtelliJ was also suggesting not to use OPtional here.
Thanks